src/employeeProfiles.js

```diff
@@ -71,22 +71,35 @@ const GetActiveContract = () => {
 
 const GetVacationsAndAvailableDays = () => {
   const employeeId = useGetRecordId();
-  const { data: vacationDetailData } = useGetManyReference("vacations", {
+  const { data: vacations } = useGetManyReference("vacations", {
     target: "employeeId",
     id: employeeId,
   });
 
   let vacationAvailableDays = 0;
-  if (vacationDetailData) {
-    vacationDetailData.forEach((v) => {
-      vacationAvailableDays += v.credit;
-      vacationAvailableDays -= v.debit;
-      v.remainingDays = v.credit - v.debit;
-    });
+  const sumPerYear = {};
+  if (vacations) {
+    for (const element of vacations) {
+      vacationAvailableDays += element.credit;
+      vacationAvailableDays -= element.debit;
+      const { year, credit, debit } = element;
+      if (!sumPerYear[year]) {
+        sumPerYear[year] = { credit: 0, debit: 0 };
+      }
+      sumPerYear[year].credit += credit;
+      sumPerYear[year].debit += debit || 0;
```

Comment:
why not set the remainingDays here? 

```
      if (!sumPerYear[year]) {
        sumPerYear[year] = { remainingDays: 0 };
      }
      sumPerYear[year].remainingDays += credit;
      sumPerYear[year].remainingDays -= debit || 0;
```

Reply:
first we need to get the credit and debit sum to calculate the remaining days